[Bug]: test_compact_during_insert failed

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

```
[2024-05-22T04:28:36.103Z] =================================== FAILURES ===================================
[2024-05-22T04:28:36.103Z] ______________ TestCompactionOperation.test_compact_during_insert ______________
[2024-05-22T04:28:36.103Z] [gw0] linux -- Python 3.8.17 /usr/local/bin/python3
[2024-05-22T04:28:36.103Z] [gw0] linux -- Python 3.8.17 /usr/local/bin/python3[gw0] linux -- Python 3.8.17 /usr/local/bin/python3
[2024-05-22T04:28:36.103Z] 
[2024-05-22T04:28:36.103Z] self = <test_compaction.TestCompactionOperation object at 0x7fdbd02a4490>
[2024-05-22T04:28:36.103Z] 
[2024-05-22T04:28:36.103Z]     @pytest.mark.tags(CaseLabel.L1)
[2024-05-22T04:28:36.103Z]     def test_compact_during_insert(self):
[2024-05-22T04:28:36.103Z]         """
[2024-05-22T04:28:36.103Z]         target: test compact during insert and flush
[2024-05-22T04:28:36.104Z]         method: 1.insert entities into multi segments
[2024-05-22T04:28:36.104Z]                 2.start a thread to load and search
[2024-05-22T04:28:36.104Z]                 3.compact collection
[2024-05-22T04:28:36.104Z]         expected: Search and compact both successfully
[2024-05-22T04:28:36.104Z]         """
[2024-05-22T04:28:36.104Z]         collection_w = self.collection_insert_multi_segments_one_shard(prefix, nb_of_segment=ct.default_nb,
[2024-05-22T04:28:36.104Z]                                                                        is_dup=False)
[2024-05-22T04:28:36.104Z]         collection_w.create_index(ct.default_float_vec_field_name, ct.default_index)
[2024-05-22T04:28:36.104Z]         log.debug(collection_w.index())
[2024-05-22T04:28:36.104Z]         df = cf.gen_default_dataframe_data(start=ct.default_nb*2)
[2024-05-22T04:28:36.104Z]     
[2024-05-22T04:28:36.104Z]         def do_flush():
[2024-05-22T04:28:36.104Z]             collection_w.insert(df)
[2024-05-22T04:28:36.104Z]             log.debug(collection_w.num_entities)
[2024-05-22T04:28:36.104Z]     
[2024-05-22T04:28:36.104Z]         # compact during insert
[2024-05-22T04:28:36.104Z]         t = threading.Thread(target=do_flush, args=())
[2024-05-22T04:28:36.104Z]         t.start()
[2024-05-22T04:28:36.104Z]         collection_w.compact()
[2024-05-22T04:28:36.104Z]         collection_w.wait_for_compaction_completed()
[2024-05-22T04:28:36.104Z]         collection_w.get_compaction_plans()
[2024-05-22T04:28:36.104Z]         t.join()
[2024-05-22T04:28:36.104Z]     
[2024-05-22T04:28:36.104Z]         # waitting for new segment index and compact
[2024-05-22T04:28:36.104Z]         index_cost = 240
[2024-05-22T04:28:36.104Z]         start = time()
[2024-05-22T04:28:36.104Z]         while True:
[2024-05-22T04:28:36.104Z]             sleep(10)
[2024-05-22T04:28:36.104Z]             collection_w.load()
[2024-05-22T04:28:36.104Z]             # new segment compacted
[2024-05-22T04:28:36.104Z]             seg_info = self.utility_wrap.get_query_segment_info(collection_w.name)[0]
[2024-05-22T04:28:36.104Z]             if len(seg_info) == 2:
[2024-05-22T04:28:36.104Z]                 break
[2024-05-22T04:28:36.104Z]             end = time()
[2024-05-22T04:28:36.104Z]             collection_w.release()
[2024-05-22T04:28:36.104Z]             if end - start > index_cost:
[2024-05-22T04:28:36.104Z]                 raise MilvusException(1, f"Waiting more than {index_cost}s for the new segment indexed")
[2024-05-22T04:28:36.104Z]     
[2024-05-22T04:28:36.104Z]         # compact new segment
[2024-05-22T04:28:36.104Z]         collection_w.compact()
[2024-05-22T04:28:36.104Z]         collection_w.wait_for_compaction_completed()
[2024-05-22T04:28:36.104Z]         collection_w.get_compaction_plans()
[2024-05-22T04:28:36.104Z]     
[2024-05-22T04:28:36.104Z]         # waitting for new segment index and compact
[2024-05-22T04:28:36.104Z]         compact_cost = 180
[2024-05-22T04:28:36.104Z]         start = time()
[2024-05-22T04:28:36.104Z]         while True:
[2024-05-22T04:28:36.104Z]             sleep(1)
[2024-05-22T04:28:36.104Z]             collection_w.load()
[2024-05-22T04:28:36.104Z]             # new segment compacted
[2024-05-22T04:28:36.104Z]             seg_info = self.utility_wrap.get_query_segment_info(collection_w.name)[0]
[2024-05-22T04:28:36.104Z]             if len(seg_info) == 1:
[2024-05-22T04:28:36.104Z]                 break
[2024-05-22T04:28:36.104Z]             end = time()
[2024-05-22T04:28:36.104Z]             collection_w.release()
[2024-05-22T04:28:36.104Z]             if end - start > compact_cost:
[2024-05-22T04:28:36.104Z]                 raise MilvusException(1, f"Waiting more than {compact_cost}s for the new target segment to load")
[2024-05-22T04:28:36.104Z]     
[2024-05-22T04:28:36.104Z]         # search
[2024-05-22T04:28:36.104Z]         search_res, _ = collection_w.search([df[ct.default_float_vec_field_name][0]],
[2024-05-22T04:28:36.104Z]                                             ct.default_float_vec_field_name,
[2024-05-22T04:28:36.104Z]                                             ct.default_search_params, ct.default_limit)
[2024-05-22T04:28:36.104Z] >       assert len(search_res[0]) == ct.default_limit
[2024-05-22T04:28:36.104Z] E       AssertionError: assert 9 == 10
[2024-05-22T04:28:36.104Z] E        +  where 9 = len(['id: 4000, distance: 1.0000474452972412, entity: {}', 'id: 3922, distance: 0.824863851070404, entity: {}', 'id: 4480, distance: 0.8140870928764343, entity: {}', 'id: 5396, distance: 0.813873827457428, entity: {}', 'id: 1162, distance: 0.8129059672355652, entity: {}', 'id: 1055, distance: 0.8105636835098267, entity: {}', 'id: 93, distance: 0.8103493452072144, entity: {}', 'id: 1262, distance: 0.8090896606445312, entity: {}', 'id: 5501, distance: 0.8077622056007385, entity: {}'])
[2024-05-22T04:28:36.104Z] E        +  and   10 = ct.default_limit
[2024-05-22T04:28:36.104Z] 
[2024-05-22T04:28:36.104Z] testcases/test_compaction.py:1209: AssertionError
[2024-05-22T04:28:36.104Z] ------------------------------ Captured log setup ------------------------------
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - INFO - ci_test]: *********************************** setup *********************************** (client_base.py:44)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - INFO - ci_test]: [setup_method] Start setup test case test_compact_during_insert. (client_base.py:45)
[2024-05-22T04:28:36.104Z] ------------------------------ Captured log call -------------------------------
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [FieldSchema] args: ['int64', <DataType.INT64: 5>, ''], kwargs: {'is_primary': False} (api_request.py:62)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : {'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>}  (api_request.py:37)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [FieldSchema] args: ['varchar', <DataType.VARCHAR: 21>, ''], kwargs: {'max_length': 65535, 'is_primary': False} (api_request.py:62)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : {'name': 'varchar', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params': {'max_length': 65535}}  (api_request.py:37)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [FieldSchema] args: ['float', <DataType.FLOAT: 10>, ''], kwargs: {'is_primary': False} (api_request.py:62)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : {'name': 'float', 'description': '', 'type': <DataType.FLOAT: 10>}  (api_request.py:37)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [FieldSchema] args: ['json_field', <DataType.JSON: 23>, ''], kwargs: {'is_primary': False} (api_request.py:62)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : {'name': 'json_field', 'description': '', 'type': <DataType.JSON: 23>}  (api_request.py:37)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [FieldSchema] args: ['float_vector', <DataType.FLOAT_VECTOR: 101>, ''], kwargs: {'dim': 128, 'is_primary': False} (api_request.py:62)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : {'name': 'float_vector', 'description': '', 'type': <DataType.FLOAT_VECTOR: 101>, 'params': {'dim': 128}}  (api_request.py:37)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [CollectionSchema] args: [[{'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>}, {'name': 'float', 'description': '', 'type': <DataType.FLOAT: 10>}, {'name': 'varchar', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params': {'max_length': 65535}}, {'name': 'json_field', 'description': '', 'type': <DataTyp......, kwargs: {'primary_field': 'int64', 'auto_id': False, 'enable_dynamic_field': False} (api_request.py:62)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : {'auto_id': False, 'description': '', 'fields': [{'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': False}, {'name': 'float', 'description': '', 'type': <DataType.FLOAT: 10>}, {'name': 'varchar', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params......  (api_request.py:37)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [Connections.has_connection] args: ['default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : False  (api_request.py:37)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [Connections.connect] args: ['default', '', '', 'default', ''], kwargs: {'host': 'ms-33255-1-pr-milvus.milvus-ci', 'port': '19530'} (api_request.py:62)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [Collection] args: ['compact_QnT9q36H', {'auto_id': False, 'description': '', 'fields': [{'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': False}, {'name': 'float', 'description': '', 'type': <DataType.FLOAT: 10>}, {'name': 'varchar', 'description': '', 'type': <DataType......., kwargs: {'shards_num': 1, 'consistency_level': 'Strong'} (api_request.py:62)
[2024-05-22T04:28:36.104Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : <Collection>:
[2024-05-22T04:28:36.104Z] -------------
[2024-05-22T04:28:36.104Z] <name>: compact_QnT9q36H
[2024-05-22T04:28:36.104Z] <description>: 
[2024-05-22T04:28:36.105Z] <schema>: {'auto_id': False, 'description': '', 'fields': [{'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': False}, {'name': 'float', 'description': '', 'type': <DataType.FLOAT: 10>},......  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float varchar                         json_field                                       float_vector
[2024-05-22T04:28:36.105Z] 0         0     0.0       0        {'number': 0, 'float': 0.0}  [0.1614141861461224, 0.136016079591544, 0.0668...
[2024-05-22T04:28:36.105Z] 1         1     1.0       1        {'number': 1, 'float': 1.0}  [0.0963......, kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_response) : (insert count: 2000, delete count: 0, upsert count: 0, timestamp: 449931382935257090, success count: 2000, err count: 0, cost: 0)  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:26:56 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:26:58 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:26:58 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float varchar                         json_field                                       float_vector
[2024-05-22T04:28:36.105Z] 0      2000  2000.0    2000  {'number': 2000, 'float': 2000.0}  [0.05943656950879113, 0.03902033951203762, 0.0...
[2024-05-22T04:28:36.105Z] 1      2001  2001.0    2001  {'number': 2001, 'float': 2001.0}  [0.0361......, kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:26:59 - DEBUG - ci_test]: (api_response) : (insert count: 2000, delete count: 0, upsert count: 0, timestamp: 449931383669522433, success count: 2000, err count: 0, cost: 0)  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:26:59 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:02 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:02 - DEBUG - ci_test]: (api_request)  : [Collection.create_index] args: ['float_vector', {'index_type': 'IVF_SQ8', 'metric_type': 'COSINE', 'params': {'nlist': 128}}, 1200], kwargs: {'index_name': ''} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:03 - DEBUG - ci_test]: (api_response) : Status(code=0, message=)  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:03 - DEBUG - ci_test]: (api_request)  : [Collection.index] args: [], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:03 - DEBUG - ci_test]: (api_response) : <pymilvus.orm.index.Index object at 0x7fdb88538e50>  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:03 - DEBUG - ci_test]: (<pymilvus.orm.index.Index object at 0x7fdb88538e50>, True) (test_compaction.py:1155)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:03 - DEBUG - ci_test]: (api_request)  : [Collection.compact] args: [180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:03 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float varchar                         json_field                                       float_vector
[2024-05-22T04:28:36.105Z] 0      4000  4000.0    4000  {'number': 4000, 'float': 4000.0}  [0.09877990860727036, 0.13545430313953316, 0.0...
[2024-05-22T04:28:36.105Z] 1      4001  4001.0    4001  {'number': 4001, 'float': 4001.0}  [0.1271......, kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:03 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:03 - DEBUG - ci_test]: (api_response) : (insert count: 2000, delete count: 0, upsert count: 0, timestamp: 449931384770527234, success count: 2000, err count: 0, cost: 0)  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:03 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:06 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:06 - DEBUG - ci_test]: 6000 (test_compaction.py:1160)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:07 - DEBUG - ci_test]: (api_request)  : [Collection.get_compaction_plans] args: [180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:07 - DEBUG - ci_test]: (api_response) : 
[2024-05-22T04:28:36.105Z] Compaction Plans:
[2024-05-22T04:28:36.105Z]  - compaction id: 449930837773559751
[2024-05-22T04:28:36.105Z]  - state: Completed
[2024-05-22T04:28:36.105Z]  - plans: [
[2024-05-22T04:28:36.105Z] Plan:
[2024-05-22T04:28:36.105Z]  - sources: [449930837773559641, 449930837773559161]
[2024-05-22T04:28:36.105Z]  - target: 449930837771183411
[2024-05-22T04:28:36.105Z] ]
[2024-05-22T04:28:36.105Z]    (api_request.py:37)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:17 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.105Z] [2024-05-22 04:27:21 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:21 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:21 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837771183411
[2024-05-22T04:28:36.106Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.106Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.106Z] num_rows: 4000
[2024-05-22T04:28:36.106Z] index_name: "float_vector"
[2024-05-22T04:28:36.106Z] indexID: 449930837773559728
[2024-05-22T04:28:36.106Z] state: Sealed
[2024-05-22T04:28:36.106Z] nodeIds: 1
[2024-05-22T04:28:36.106Z] , segmentID: 449930837773559753
[2024-05-22T04:28:36.106Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.106Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.106Z] num_rows: 20......  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:21 - DEBUG - ci_test]: (api_request)  : [Collection.compact] args: [180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:21 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:25 - DEBUG - ci_test]: (api_request)  : [Collection.get_compaction_plans] args: [180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:25 - DEBUG - ci_test]: (api_response) : 
[2024-05-22T04:28:36.106Z] Compaction Plans:
[2024-05-22T04:28:36.106Z]  - compaction id: 449930837773561189
[2024-05-22T04:28:36.106Z]  - state: Completed
[2024-05-22T04:28:36.106Z]  - plans: [
[2024-05-22T04:28:36.106Z] Plan:
[2024-05-22T04:28:36.106Z]  - sources: [449930837773559753, 449930837771183411]
[2024-05-22T04:28:36.106Z]  - target: 449930837771183624
[2024-05-22T04:28:36.106Z] ]
[2024-05-22T04:28:36.106Z]    (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:26 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:26 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:26 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:26 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837771183411
[2024-05-22T04:28:36.106Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.106Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.106Z] num_rows: 4000
[2024-05-22T04:28:36.106Z] index_name: "float_vector"
[2024-05-22T04:28:36.106Z] indexID: 449930837773559728
[2024-05-22T04:28:36.106Z] state: Sealed
[2024-05-22T04:28:36.106Z] nodeIds: 1
[2024-05-22T04:28:36.106Z] , segmentID: 449930837773559753
[2024-05-22T04:28:36.106Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.106Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.106Z] num_rows: 20......  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:26 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:27 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:28 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:32 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:32 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:32 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837771183411
[2024-05-22T04:28:36.106Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.106Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.106Z] num_rows: 4000
[2024-05-22T04:28:36.106Z] index_name: "float_vector"
[2024-05-22T04:28:36.106Z] indexID: 449930837773559728
[2024-05-22T04:28:36.106Z] state: Sealed
[2024-05-22T04:28:36.106Z] nodeIds: 1
[2024-05-22T04:28:36.106Z] , segmentID: 449930837773559753
[2024-05-22T04:28:36.106Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.106Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.106Z] num_rows: 20......  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:32 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:33 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:34 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:38 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.106Z] [2024-05-22 04:27:38 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:38 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837773559753
[2024-05-22T04:28:36.107Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.107Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.107Z] num_rows: 2000
[2024-05-22T04:28:36.107Z] index_name: "float_vector"
[2024-05-22T04:28:36.107Z] indexID: 449930837773559728
[2024-05-22T04:28:36.107Z] state: Sealed
[2024-05-22T04:28:36.107Z] nodeIds: 1
[2024-05-22T04:28:36.107Z] , segmentID: 449930837771183411
[2024-05-22T04:28:36.107Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.107Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.107Z] num_rows: 40......  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:38 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:39 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:40 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:43 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:43 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:43 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837773559753
[2024-05-22T04:28:36.107Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.107Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.107Z] num_rows: 2000
[2024-05-22T04:28:36.107Z] index_name: "float_vector"
[2024-05-22T04:28:36.107Z] indexID: 449930837773559728
[2024-05-22T04:28:36.107Z] state: Sealed
[2024-05-22T04:28:36.107Z] nodeIds: 1
[2024-05-22T04:28:36.107Z] , segmentID: 449930837771183411
[2024-05-22T04:28:36.107Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.107Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.107Z] num_rows: 40......  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:43 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:44 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:45 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:50 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:50 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:50 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837771183411
[2024-05-22T04:28:36.107Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.107Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.107Z] num_rows: 4000
[2024-05-22T04:28:36.107Z] index_name: "float_vector"
[2024-05-22T04:28:36.107Z] indexID: 449930837773559728
[2024-05-22T04:28:36.107Z] state: Sealed
[2024-05-22T04:28:36.107Z] nodeIds: 1
[2024-05-22T04:28:36.107Z] , segmentID: 449930837773559753
[2024-05-22T04:28:36.107Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.107Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.107Z] num_rows: 20......  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:50 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:51 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:52 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:55 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:55 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:55 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837771183411
[2024-05-22T04:28:36.107Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.107Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.107Z] num_rows: 4000
[2024-05-22T04:28:36.107Z] index_name: "float_vector"
[2024-05-22T04:28:36.107Z] indexID: 449930837773559728
[2024-05-22T04:28:36.107Z] state: Sealed
[2024-05-22T04:28:36.107Z] nodeIds: 1
[2024-05-22T04:28:36.107Z] , segmentID: 449930837773559753
[2024-05-22T04:28:36.107Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.107Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.107Z] num_rows: 20......  (api_request.py:37)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:55 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.107Z] [2024-05-22 04:27:56 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:27:57 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:00 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:00 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:00 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837771183411
[2024-05-22T04:28:36.108Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.108Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.108Z] num_rows: 4000
[2024-05-22T04:28:36.108Z] index_name: "float_vector"
[2024-05-22T04:28:36.108Z] indexID: 449930837773559728
[2024-05-22T04:28:36.108Z] state: Sealed
[2024-05-22T04:28:36.108Z] nodeIds: 1
[2024-05-22T04:28:36.108Z] , segmentID: 449930837773559753
[2024-05-22T04:28:36.108Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.108Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.108Z] num_rows: 20......  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:00 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:01 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:02 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:05 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:05 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:05 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837773559753
[2024-05-22T04:28:36.108Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.108Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.108Z] num_rows: 2000
[2024-05-22T04:28:36.108Z] index_name: "float_vector"
[2024-05-22T04:28:36.108Z] indexID: 449930837773559728
[2024-05-22T04:28:36.108Z] state: Sealed
[2024-05-22T04:28:36.108Z] nodeIds: 1
[2024-05-22T04:28:36.108Z] , segmentID: 449930837771183411
[2024-05-22T04:28:36.108Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.108Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.108Z] num_rows: 40......  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:05 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:06 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:07 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:10 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:10 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:10 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837773559753
[2024-05-22T04:28:36.108Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.108Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.108Z] num_rows: 2000
[2024-05-22T04:28:36.108Z] index_name: "float_vector"
[2024-05-22T04:28:36.108Z] indexID: 449930837773559728
[2024-05-22T04:28:36.108Z] state: Sealed
[2024-05-22T04:28:36.108Z] nodeIds: 1
[2024-05-22T04:28:36.108Z] , segmentID: 449930837771183411
[2024-05-22T04:28:36.108Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.108Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.108Z] num_rows: 40......  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:10 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:11 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:12 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:15 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:15 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:15 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837773559753
[2024-05-22T04:28:36.108Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.108Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.108Z] num_rows: 2000
[2024-05-22T04:28:36.108Z] index_name: "float_vector"
[2024-05-22T04:28:36.108Z] indexID: 449930837773559728
[2024-05-22T04:28:36.108Z] state: Sealed
[2024-05-22T04:28:36.108Z] nodeIds: 1
[2024-05-22T04:28:36.108Z] , segmentID: 449930837771183411
[2024-05-22T04:28:36.108Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.108Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.108Z] num_rows: 40......  (api_request.py:37)
[2024-05-22T04:28:36.108Z] [2024-05-22 04:28:15 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:16 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:17 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:20 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:20 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:20 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837771183411
[2024-05-22T04:28:36.109Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.109Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.109Z] num_rows: 4000
[2024-05-22T04:28:36.109Z] index_name: "float_vector"
[2024-05-22T04:28:36.109Z] indexID: 449930837773559728
[2024-05-22T04:28:36.109Z] state: Sealed
[2024-05-22T04:28:36.109Z] nodeIds: 1
[2024-05-22T04:28:36.109Z] , segmentID: 449930837773559753
[2024-05-22T04:28:36.109Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.109Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.109Z] num_rows: 20......  (api_request.py:37)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:20 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:21 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:22 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:25 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:25 - DEBUG - ci_test]: (api_request)  : [get_query_segment_info] args: ['compact_QnT9q36H', 20, 'default'], kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:25 - DEBUG - ci_test]: (api_response) : [segmentID: 449930837771183624
[2024-05-22T04:28:36.109Z] collectionID: 449930837773359149
[2024-05-22T04:28:36.109Z] partitionID: 449930837773359150
[2024-05-22T04:28:36.109Z] num_rows: 6000
[2024-05-22T04:28:36.109Z] index_name: "float_vector"
[2024-05-22T04:28:36.109Z] indexID: 449930837773559728
[2024-05-22T04:28:36.109Z] state: Sealed
[2024-05-22T04:28:36.109Z] nodeIds: 1
[2024-05-22T04:28:36.109Z] ]  (api_request.py:37)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:25 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.09877990860727036, 0.13545430313953316, 0.04072070215911992, 0.060953618624993926, 0.07263572185632518, 0.0067934192610247255, 0.14136217580049057, 0.10421176585288708, 0.02024821795463133, 0.04715504434250422, 0.08635936579143884, 0.09014656712897334, 0.1178421091919146, 0.1482627674773467, 0......., kwargs: {} (api_request.py:62)
[2024-05-22T04:28:36.109Z] [2024-05-22 04:28:25 - DEBUG - ci_test]: (api_response) : data: ["['id: 4000, distance: 1.0000474452972412, entity: {}', 'id: 3922, distance: 0.824863851070404, entity: {}', 'id: 4480, distance: 0.8140870928764343, entity: {}', 'id: 5396, distance: 0.813873827457428, entity: {}', 'id: 1162, distance: 0.8129059672355652, entity: {}', 'id: 1055, distance: 0.......  (api_request.py:37)
```

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

--- a/internal/storage/serde.go
+++ b/internal/storage/serde.go
@@ -538,8 +538,9 @@ func (deser *DeserializeReader[T]) Next() error {
 		deser.pos = 0
 		deser.rec = deser.rr.Record()
 
-		// allocate new slice preventing overwrite previous batch
-		deser.values = make([]T, deser.rec.Len())
+		if deser.values == nil || len(deser.values) != deser.rec.Len() {
+			deser.values = make([]T, deser.rec.Len())
+		}
 		if err := deser.deserializer(deser.rec, deser.values); err != nil {
 			return err
 		}

